sources/packages/backend/apps/api/src/route-controllers/designation-agreement/designation-agreement.institutions.controller.ts

```diff
@@ -81,6 +83,15 @@ export class DesignationAgreementInstitutionsController extends BaseController {
         "User does not have the rights to create a designation agreement.",
       );
     }
+
+    // Check if institution is private or public and append it to the payload
+    const isInstitutionPrivate =
+      await this.institutionControllerService.checkIfInstitutionIsPrivate(
+        userToken.authorizations.institutionId,
+      );
+
+    payload.isBCPrivate = isInstitutionPrivate;
```

Comment:
Just a suggestion, creating a separate variable sometimes makes the code better to read, for this case I would not mind having it directly associated.
```ts
payload.isBCPrivate = await this.institutionControllerService.checkIfInstitutionIsPrivate(
        userToken.authorizations.institutionId,
      );
```